lib/overcommit/configuration.rb

```diff
@@ -264,8 +265,10 @@ def ad_hoc_hook?(hook_context, hook_name)
     def built_in_hook?(hook_context, hook_name)
       hook_name = Overcommit::Utils.snake_case(hook_name)
 
-      File.exist?(File.join(Overcommit::HOME, 'lib', 'overcommit', 'hook',
-                            hook_context.hook_type_name, "#{hook_name}.rb"))
+      $LOAD_PATH.any? do |dir|
```

Comment:
I'm wary of this approach because it bypasses the standard mechanism for loading hook plugins, which ensures the source code of the plugins is properly signed.

By allowing code loading this way, an attacker can easily inject malicious code by simply adding a gem to the Gemfile specified in the `.overcommit.yml` `gemfile` option. We need to have a more explicit way of specifying which code is loaded so that Overcommit can properly sign the hooks to protect against this attack.